FSW/src/util/memUtil.cpp

```diff
@@ -214,10 +214,49 @@ bool saveBuffer(int &index){
     // TODO: may need to seek position to end of file before writing again
     //file.seek(BUFFERSIZE);  
 
-    //TODO: decide on timestamp format and append it to file
-    //file.write(timestamp);
+    // compute and write timestamp
+    unsigned long timestamp = calcTimestamp();
+    file.write(&timestamp, sizeof(timestamp));
 
     index = 0; // reset index to start of array since we have saved the buffer
 
     return status; // return whether or not file was successfully created
+}
+
+/* - - - - - - calcTimestamp - - - - - - *
+ * Belongs to Science Memory Handling Module
+ *  
+ * Usage:
+ *  computes the relative time between teensy startup and when this file is saved
+ *  returns relative time be appended to the end of the file   
+ *  NOTE: in the file, only the timestamp of the final data point will be 
+ *        available. Use GSW and the sampling rate of FSW to backsolve for the 
+ *        timestamp of all preceding data points
+ * 
+ * FOR FUTURE TEAMS:
+ * this year's time is relative since we do not have a bus clock signal
+ *   however, it may be desirable to downlink the times in UTC or another standardized
+ *   time so that you do not have to keep track of the time that the teensy powered on
+ * By piping in a clock signal from the bus, you could potentially use the 
+ *  teensy time library to append timestamp
+ *  https://www.pjrc.com/teensy/td_libs_Time.html
+ *  might be useful to convert timestamp to ISO 8601 timekeeping standard for storing to file
+ *       ([YYYY]-[MM]-[DD]T[hh]:[mm]:[ss].[sss] where [sss] is subseconds)
+ * 
+ * Inputs:
+ *  none
+ *  
+ * Outputs:
+ *  relative timestamp
+ */
+unsigned long calcTimestamp(){
+    unsigned long currentTimeRelative = millis(); // milliseconds    
```

Comment:
I'm guessing all this hooplah is just a nicety for future teams?

Reply:
yes hopefully they understand that they need to do something here based on the paragraph of comments